google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -399,6 +401,23 @@ public void run() {
     }
   }
 
+  /** Compute the ideal deadline, set subsequent modacks to this deadline, and return it. */
+  @InternalApi
+  int computeAndSetDeadlineSeconds() {
+    long secLong = ackLatencyDistribution.getNthPercentile(PERCENTILE_FOR_ACK_DEADLINE_UPDATES);
+    int sec = Ints.saturatedCast(secLong);
+
+    // Use Ints.constrainToRange when we get guava 21.
+    if (sec < Subscriber.MIN_ACK_DEADLINE_SECONDS) {
+      sec = Subscriber.MIN_ACK_DEADLINE_SECONDS;
+    } else if (sec > Subscriber.MAX_ACK_DEADLINE_SECONDS) {
+      sec = Subscriber.MAX_ACK_DEADLINE_SECONDS;
+    }
+
+    messageDeadlineSeconds.set(sec);
```

Comment:
It seems like this should either set messageDeadlineSeconds or return the variable, callers should be responsible for dealing with the return value if they need it and they can also set messageDeadlineSeconds.

Reply:
Done